backend/routes/offline-payment.js

```diff
@@ -79,11 +85,69 @@ module.exports = function (router) {
         return res.json({ success: false, message: 'Missing order data' })
       }
 
+      const paymentCode = randomstring.generate()
+
       req.body.data = encryptedData
       req.amount = 0
-      req.paymentCode = randomstring.generate()
+      req.paymentCode = paymentCode
+      req.paymentStatus = OrderPaymentStatuses.Pending
+
+      // Insert a new row in Transaction table
+      await Transaction.create({
+        shopId: shop.id,
+        networkId: network.networkId,
+        type: TransactionTypes.Payment,
+        status: TransactionStatuses.Pending,
+        listingId: shop.listingId,
+        customId: paymentCode // Record the paymentCode in the custom_id field.
+      })
+
       next()
     },
     makeOffer
   )
+
+  router.put(
+    '/offline-payments/payment-state',
+    authSellerAndShop,
+    async (req, res) => {
+      const { paymentCode, state } = req.body
+
+      const tx = await Transaction.findOne({
+        where: {
+          customId: paymentCode,
+          shopId: req.shop.id
+        }
+      })
+
+      const order = await Order.findOne({
+        where: {
+          shopId: req.shop.id,
+          paymentCode
+        }
+      })
+
+      if (!tx || !order) {
+        return res.status(200).send({
+          reason: 'Invalid payment code'
+        })
+      }
+
+      const orderStateToTxState = {
+        [OrderPaymentStatuses.Paid]: TransactionStatuses.Confirmed,
```

Comment:
What about simply returning order.paymentStatus rather than mapping it into a transaction status?